Source/SpectrumAnalysis/KTSeqTrackFinder.cc

```diff
@@ -0,0 +1,539 @@
+        }
+        if (node->has("max-frequency"))
+        {
+        	SetMaxFrequency(node->get_value< double >("max-frequency"));
+        }
+        if (node->has("min-bin"))
+        {
+            SetMinBin(node->get_value< unsigned >("min-bin"));
+            SetCalculateMinBin(false);
+        }
+        if (node->has("max-bin"))
+        {
+            SetMaxBin(node->get_value< unsigned >("max-bin"));
+            SetCalculateMaxBin(false);
+        }
+        if (node->has("trimming-factor"))
+        {
+            SetTrimmingFactor(node->get_value<double>("trimming-factor"));
+        }
+        if (node->has("half-line-width"))
+        {
+            SetLinePowerWidth(node->get_value<unsigned>("half-line-width"));
+        }
+        if (node->has("time-gap-tolerance"))
+        {
+            SetTimeGapTolerance(node->get_value<double>("time-gap-tolerance"));
+        }
+        if (node->has("minimum-line-bin-distance"))
+		{
+        	SetMinFreqBinDistance(node->get_value<double>("minimum-line-bin-distance"));
+		}
+        if (node->has("search-radius"))
+        {
+        	SetSearchRadius(node->get_value<int>("search-radius"));
+        }
+        if (node->has("converge-delta"))
+        {
+            SetConvergeDelta(node->get_value<int>("converge_delta"));
+        }
+        if (node->has("frequency-acceptance"))
+        {
+            SetFrequencyAcceptance(node->get_value<double>("frequency-acceptance"));
+        }
+        if (node->has("initial-slope"))
+        {
+            SetInitialSlope(node->get_value<double>("initial-slope"));
+        }
+        if (node->has("min-points"))
+        {
+            SetMinPoints(node->get_value<int>("min-points"));
+        }
+        if (node->has("min-slope"))
+        {
+            SetMinSlope(node->get_value<int>("min-slope"));
+        }
+        if (node->has("apply-power-cut"))
+        {
+            SetApplyPowerCut(node->get_value<bool>("apply-power-cut"));
+        }
+        if (node->has("apply-point-density-cut"))
+        {
+            SetApplyDensityCut(node->get_value<bool>("apply-point-density-cut"));
+        }
+        if (node->has("power-threshold"))
+        {
+            SetPowerThreshold(node->get_value<double>("power-threshold"));
+        }
+        if (node->has("point-density-threshold"))
+        {
+            SetDensityThreshold(node->get_value<double>("point-density-threshold"));
+        }
+
+
+        return true;
+    }
+
+    bool KTSeqTrackFinder::SetPreCalcGainVar(KTGainVariationData& gvData)
+    {
+        fGVData = gvData;
+        return true;
+    }
+
+    bool KTSeqTrackFinder::RunSequentialTrackFinding(KTSliceHeader& slHeader, KTPowerSpectrumData& spectrum)
+    {
+        this->PointLineAssignment(slHeader, spectrum, fGVData);
+        return true;
+    }
+
+    bool KTSeqTrackFinder::PointLineAssignment(KTSliceHeader& slHeader, KTPowerSpectrumData& spectrum, KTGainVariationData& gvData)
```

Comment:
Instead of doing the discrimination here, can we use the Variable Spectrum Discriminator and take discriminated points as the input to this processor?

Reply:
No, because the discriminated points are taken as seeds. The algorithm then takes the surrounding points into consideration to calculate a "true" frequency and amplitude of the track in this time slice.

Reply:
If the processor takes both the discriminated points and the full spectrum as input, then could you still use the discriminated points as seeds?

Reply:
In principle yes, but I am not sure how. Because I would still need the slice header and the power spectrum. Could we make a ThreeType Slot?

Reply:
Of course!  In Nymph 2.0 you'll be able to use an arbitrary number of arguments, but we don't have that quite yet, so I can take care of adding it explicitly for the time being.